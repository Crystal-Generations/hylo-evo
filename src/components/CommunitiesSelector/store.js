import { createSelector } from 'reselect'
import sampleCommunities from './sampleCommunities'

const FIND_COMMUNITIES = 'hyloEditor/FIND_COMMUNITIES'
const CLEAR_COMMUNITIES = 'hyloEditor/CLEAR_COMMUNITIES'

// Action Creators

export function findCommunities (searchText) {
  return {
    type: FIND_COMMUNITIES,
    payload: { searchText }
  }
}

export function clearCommunities (searchText) {
  return { type: CLEAR_COMMUNITIES }
}

// Reducer

const defaultState = {
  communitiesResults: sampleCommunities
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_COMMUNITIES:
      // filter(i =>
      //   String(i
      //     .get('name')
      //     .replace(/\s+/g, '')
      //     .toLowerCase()
      //     .indexOf(lowerSearch) !== -1
      //   )
      // )
      // .take(5)
      return {...state, communitiesResults: filter(payload.searchText)(sampleCommunities)}
    case CLEAR_COMMUNITIES:
      return {...state, communitiesResults: []}
    default:
      return state
  }
}

// Selectors

export const getCommunitiesResults = createSelector(
  (state) => state.CommunitiesSelector,
  (state, props) => state.communitiesResults
)
