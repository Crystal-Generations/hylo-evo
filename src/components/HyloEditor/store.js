import { createSelector } from 'reselect'
import { fromJS } from 'immutable'
import sampleMentions from './sampleMentions'
import mentionPlugin from 'draft-js-mention-plugin'
import hashtagPlugin from './hashtagPlugin'

const defaultMentionsSuggestionFilter = mentionPlugin.defaultSuggestionsFilter
const defaultHashtagSuggestionFilter = hashtagPlugin.defaultSuggestionsFilter

const FIND_MENTIONS = 'hyloEditor/FIND_MENTIONS'
const CLEAR_MENTIONS = 'hyloEditor/CLEAR_MENTIONS'
const FIND_HASHTAGS = 'hyloEditor/FIND_HASHTAGS'
const CLEAR_HASHTAGS = 'hyloEditor/CLEAR_HASHTAGS'

// Action Creators

export function findMentions (searchText) {
  return {
    type: FIND_MENTIONS,
    payload: { searchText }
  }
}

export function clearMentions (searchText) {
  return { type: CLEAR_MENTIONS }
}

export function findHashtags (searchText) {
  return {
    type: FIND_HASHTAGS,
    payload: { searchText }
  }
}

export function clearHashtags (searchText) {
  return { type: CLEAR_HASHTAGS }
}

// Reducer

export default function reducer (state = {mentionResults: fromJS([])}, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_MENTIONS:
      return {...state, mentionResults: defaultMentionsSuggestionFilter(payload.searchText, fromJS(sampleMentions))}
    case CLEAR_MENTIONS:
      return {...state, mentionResults: fromJS([])}
    case FIND_HASHTAGS:
      return {...state, mentionResults: defaultHashtagSuggestionFilter(payload.searchText, fromJS(['test', 'test2']))}
    case CLEAR_HASHTAGS:
      return {...state, mentionResults: fromJS([])}
    default:
      return state
  }
}

// Selectors

export const getMentionResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.mentionResults
)

export const getHashtagResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.hasTagResults
)
