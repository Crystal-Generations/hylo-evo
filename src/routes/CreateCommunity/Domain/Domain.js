import React, { Component } from 'react'
import '../CreateCommunity.scss'
import ModalSidebar from 'components/ModalSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from 'components/ModalFooter'

export default class Domain extends Component {
  constructor (props) {
    super(props)
    this.state = {
      'communityDomain': ''
    }
  }

  formatDomainWithUrl (communityDomain) {
    let c = communityDomain.replace('hylo.com/c/', '').replace('hylo.com/c', '')
    if (c !== '') {
      c = 'hylo.com/c/' + c
    }
    return c
  }

  removeUrlFromDomain (communityDomain) {
    return communityDomain.replace('hylo.com/c/', '')
  }

  handleDomainChange = (event) => {
    const communityDomain = event.target.value
    this.setState({
      communityDomain: this.formatDomainWithUrl(communityDomain)
    })
  }

  submit = () => {
    const communityDomain = this.state.communityDomain
    this.props.addCommunityDomain(this.removeUrlFromDomain(communityDomain))
    this.props.goToNextStep()
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.submit()
    }
  }

  communityDomainWithUrl = () => {
    return `${this.props.communityDomain}.hylo.com`
  }

  componentWillMount = () => {
    const { communityDomain } = this.props
    if (communityDomain) this.setState({communityDomain: communityDomain})
  }
  render () {
    return <div styleName='flex-wrapper'>
      <ModalSidebar
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header='Choose an address for your community'
        body='Your URL is the address that people will use to access your community online.'
        secondParagraph='The shorter the better!'
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 2/3</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>{ this.state.communityDomain && 'Choose a domain name'}</span>
          <TextInput
            type='text'
            name='community-name'
            value={this.state.communityDomain}
            onChange={this.handleDomainChange}
            theme={inputTheme}
            placeholder='Choose a domain name'
            showClearButton={false}
            onEnter={this.onEnter}
          />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.props.goToPreviousStep}
        hidePrevious={false}
        continueText={'Continue'}
        />
    </div>
  }
}

const inputTheme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}
