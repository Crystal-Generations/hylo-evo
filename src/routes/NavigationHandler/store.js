const NAVIGATE = 'NAVIGATE'
const RESET_NAVIGATION = 'RESET_NAVIGATION'

export function navigate (to) {
  return {
    type: NAVIGATE,
    payload: {to}
  }
}

export function resetNavigation (to) {
  return {type: RESET_NAVIGATION}
}

const reducer = (state = null, { type, error, payload, meta }) => {
  switch (type) {
    case NAVIGATE:
      return payload.to
    case RESET_NAVIGATION:
      return null
  }
  return state
}

export default reducer
