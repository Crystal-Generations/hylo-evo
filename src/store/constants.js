export const _PENDING = '_PENDING'

export const CREATE_COMMENT = 'CREATE_COMMENT'
export const CREATE_COMMENT_PENDING = CREATE_COMMENT + _PENDING

export const FETCH_COMMENTS = 'FETCH_COMMENTS'
export const FETCH_FEED_ITEMS = 'FETCH_FEED_ITEMS'
export const FETCH_POST = 'FETCH_POST'
export const FETCH_POSTS = 'FETCH_POSTS'

export const FETCH_CURRENT_USER = 'FETCH_CURRENT_USER'
export const TOGGLE_COMMUNITIES_DRAWER = 'TOGGLE_COMMUNITIES_DRAWER'

export const EXTRACT_MODEL = 'EXTRACT_MODEL'
