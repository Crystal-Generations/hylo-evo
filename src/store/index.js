import React from 'react'
import { createStore, applyMiddleware, combineReducers, compose } from 'redux'
import { Provider } from 'react-redux'
import { ConnectedRouter, routerReducer, routerMiddleware } from 'react-router-redux'
import createHistory from 'history/createBrowserHistory'
import { compact, flatten } from 'lodash'
import reducers from '../reducers'
import routes from '../routes'

const history = createHistory()

// const middleware = compact(flatten([
//   routerMiddleware(history)
// ]))

const middleware = routerMiddleware(history)

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(
  combineReducers({
    // ...reducers,
    router: routerReducer
  }),
  composeEnhancers(
    applyMiddleware(middleware)
  )
)

const app = <Provider store={store}>
  <ConnectedRouter history={history}>
    {routes}
  </ConnectedRouter>
</Provider>

export default app
