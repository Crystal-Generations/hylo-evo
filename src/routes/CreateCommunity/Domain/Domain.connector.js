import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import { addCommunityDomain, fetchCommunity } from './Domain.store'

export function mapStateToProps (state, props) {
  return {
    communityName: state.CreateCommunity.name,
    domainName: state.CreateCommunity.domain
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-community/review')),
    goToPreviousStep: () => dispatch(push('/create-community/name')),
    addCommunityDomain: (domain) => dispatch(addCommunityDomain(domain)),
    fetchCommunity: (slug) => dispatch(fetchCommunity(slug))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
