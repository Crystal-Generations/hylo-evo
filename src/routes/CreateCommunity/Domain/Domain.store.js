import { ADD_COMMUNITY_DOMAIN } from '../CreateCommunity.store'

export function addCommunityDomain (domain) {
  return {
    type: ADD_COMMUNITY_DOMAIN,
    payload: {
      domain
    }
  }
}
