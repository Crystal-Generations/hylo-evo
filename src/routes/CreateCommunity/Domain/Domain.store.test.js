import { fetchCommunityExists } from './Domain.store'

describe('Domain store', () => {
  it('should call fetchCommunityExists', () => {
    const slug = 'slug'
    expect(fetchCommunityExists(slug)).toMatchSnapshot()
  })
})
