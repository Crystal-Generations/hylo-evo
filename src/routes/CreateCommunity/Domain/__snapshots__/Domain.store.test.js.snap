// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Domain store should call fetchCommunityExists 1`] = `
Object {
  "graphql": Object {
    "query": "
        query ($slug: String) {
          communityExists (slug: $slug) {
            data
          }
        }
      ",
    "variables": Object {
      "slug": "slug",
    },
  },
  "type": "CREATE_COMMUNITY_FETCH_COMMUNITY_EXISTS",
}
`;
